packages/runtime/container-runtime/src/channelCollection.ts

```diff
@@ -1406,15 +1428,17 @@ export class ChannelCollection implements IFluidDataStoreChannel, IDisposable {
 	/**
 	 * Called by GC to retrieve the package path of a data store node with the given path.
 	 */
-	public async getDataStorePackagePath(
-		nodePath: string,
-	): Promise<readonly string[] | undefined> {
+	public getDataStorePackagePath(nodePath: string): readonly string[] | undefined {
 		// If the node belongs to a data store, return its package path. For DDSes, we return the package path of the
 		// data store that contains it.
-		// TODO why are we non null asserting here?
-		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		const context = this.contexts.get(nodePath.split("/")[1]!);
-		return (await context?.getInitialSnapshotDetails())?.pkg;
+		const dataStoreId = nodePath.split("/")[1];
+		if (dataStoreId !== undefined) {
```

Comment:
Maybe mention in the doc comment that if you pass the root path it'll return undefined?  And I guess mention the other case where it returns undefined which is if it's for a context that's not loaded yet.